datasets/wikitablequestions/wikitablequestions.py

```diff
@@ -0,0 +1,183 @@
+            # specify them. They'll be used if as_supervised=True in builder.as_dataset.
+            # supervised_keys=("sentence", "label"),
+            # Homepage of the dataset for documentation
+            homepage=_HOMEPAGE,
+            # License for the dataset if available
+            license=_LICENSE,
+            # Citation for the dataset
+            citation=_CITATION,
+        )
+
+    def _split_generators(self, dl_manager):
+        train_file = "{}-train.tsv".format(self.config.name)
+        dev_file = "{}-dev.tsv".format(self.config.name)
+        test_file = "pristine-unseen-tables.tsv"
+        # dl_manager is a datasets.download.DownloadManager that can be used to download and extract URLS
+        urls = _DATA_URL
+        root_dir = os.path.join(dl_manager.download_and_extract(urls), "WikiTableQuestions")
+        return [
+            datasets.SplitGenerator(
+                name=datasets.Split.TRAIN,
+                # These kwargs will be passed to _generate_examples
+                gen_kwargs={"main_filepath": os.path.join(root_dir, "data", train_file), "root_dir": root_dir},
+            ),
+            datasets.SplitGenerator(
+                name=datasets.Split.TEST,
+                # These kwargs will be passed to _generate_examples
+                gen_kwargs={"main_filepath": os.path.join(root_dir, "data", test_file), "root_dir": root_dir},
+            ),
+            datasets.SplitGenerator(
+                name=datasets.Split.VALIDATION,
+                # These kwargs will be passed to _generate_examples
+                gen_kwargs={"main_filepath": os.path.join(root_dir, "data", dev_file), "root_dir": root_dir},
+            ),
+        ]
+
+    def _read_table_from_file(self, table_name: str, root_dir: str):
+        def _extract_table_content(_line: str):
+            _vals = [_.replace("\n", " ").strip() for _ in _line.strip("\n").split("\t")]
+            return _vals
+
+        rows = []
+        # assert ".csv" in _wtq_table_name
+        # use the normalized table file
+        table_name = table_name.replace(".csv", ".tsv")
+        with open(os.path.join(root_dir, table_name), "r", encoding="utf8") as table_f:
+            table_lines = table_f.readlines()
+            # the first line is header
+            header = _extract_table_content(table_lines[0])
+            for line in table_lines[1:]:
+                rows.append(_extract_table_content(line))
+        return {"header": header, "rows": rows, "name": table_name}
+
+    # method parameters are unpacked from `gen_kwargs` as given in `_split_generators`
+    def _generate_examples(self, main_filepath, root_dir):
+        # The `key` is for legacy reasons (tfds) and is not important in itself, but must be unique for each example.
+        with open(main_filepath, encoding="utf-8") as f:
+            # skip the first line since it is the tsv header
+            for idx, line in enumerate(f.readlines()[1:]):
```

Comment:
> Thanks a lot ! Since the dummy data files are still >50KB each, feel free to only keep the `random-split-1` one - the others can be removed
> 
> I only have one final comment. Let me know when you're done with the dummy data and I think we can merge :)

I have tried to remove `random-split-x` data. However, I cannot pass the command `pytest tests/test_dataset_common.py::LocalDatasetTest::test_load_dataset_all_configs_wikitablequestions` after removing them. Therefore, I may not remove these dummy data.

Reply:
This test is optional ^^ Indeed for cases such as this dataset it's fine if we only test one configuration, since they're all pretty much the same

In particular we don't run the test for all configurations in the CI, we only run the test for one configuration

Reply:
I see. I will fix that.

Reply:
Fixed :)